pkg/server/sandbox_run.go

```diff
@@ -179,32 +153,29 @@ func (c *criContainerdService) RunPodSandbox(ctx context.Context, r *runtime.Run
 		}
 	}()
 
-	createOpts := &tasks.CreateTaskRequest{
-		ContainerID: id,
-		Rootfs:      rootfs,
-		// No stdin for sandbox container.
-		Stdout: stdout,
-		Stderr: stderr,
+	fifo := &containerd.FIFOSet{
+		Dir: sandboxRootDir,
+		Out: stdout,
+		Err: stderr,
 	}
-	// Create sandbox task in containerd.
-	glog.V(5).Infof("Create sandbox container (id=%q, name=%q) with options %+v.",
-		id, name, createOpts)
-	createResp, err := c.taskService.Create(ctx, createOpts)
+	task, err := container.NewTask(ctx, containerd.NewIOWithFifoSet(os.Stdin, os.Stdout, os.Stderr, fifo, false))
```

Comment:
@abhinandanpb That's not how it works.

Containerd create FIFOs for each container. Currently, we create those fifos and give the path to containerd. In fact, we could use containerd client directly, which creates fifos for us in `/tmp/containerd/XXX`. We don't really care where the FIFOs are.

We create a **Logger** for each container, which read log from the FIFO, decorate the log, and write to the CRI log file, which is `/var/log/pods/XXX`. The test is also checking this path, the test doesn't care where the FIFOs are, either.